lib/modules/datasource/docker/index.ts

```diff
@@ -656,8 +656,12 @@ export class DockerDatasource extends Datasource {
       ecrRegex.test(registryHost) || ecrPublicRegex.test(registryHost)
         ? 1000
         : 10000;
+    // Add the last parameter if currentValue is provided to optimize tag fetching for Quay
+    const isQuay = registryHost === 'https://quay.io';
     let url: string | null =
-      `${registryHost}/${dockerRepository}/tags/list?n=${limit}`;
+      isQuay && currentValue
+        ? `${registryHost}/${dockerRepository}/tags/list?n=${limit}&last=${currentValue}`
```

Comment:
It would be simpler to have a if block to append this later, including the comments describing why